designer/server/src/test/scala/pl/touk/nussknacker/ui/api/NodesApiHttpServiceBusinessSpec.scala

```diff
@@ -926,6 +939,43 @@ class NodesApiHttpServiceBusinessSpec
     }
   }
 
+  "TypingResult schemas shouldn't go out of sync with Codecs" in {
+    val typingSchema = TypingDtoSchemas.typingResult
+
+    val jsonSchema = TapirSchemaToJsonSchema(
+      typingSchema,
+      markOptionsAsNullable = true
+    ).asJson
+    val schemaStr: String = Printer.spaces2.print(jsonSchema.deepDropNullValues)
+    val jsonObject        = new JSONObject(schemaStr)
+
+    val schemaLoader = SchemaLoader
+      .builder()
+      .schemaClient(SchemaClient.classPathAwareClient())
+      .registerSchemaByURI(new URI("pl.touk.nussknacker.engine.api.typed.typing.TypingResult"), jsonObject)
+      .schemaJson(jsonObject)
+      .build()
+
+    val schema: Schema =
+      schemaLoader
+        .load()
+        .build()
+        .asInstanceOf[Schema]
+
+    for (_ <- 1 to 10) {
+      TypingResultGen.typingResultGen(EnabledTypedFeatures.All).sample match {
+        case Some(sample) =>
+          val sampleJson        = TypingResult.encoder.apply(sample)
+          val sampleStr: String = Printer.spaces2.print(sampleJson.deepDropNullValues)
+          val sampleJsonObject  = new JSONObject(sampleStr)
+
+          Try(schema.validate(sampleJsonObject)) shouldBe Success(())
```

Comment:
it'd be great if we could create a [custom matcher](https://www.scalatest.org/scaladoc/1.6.1/org/scalatest/matchers/Matcher.html) for the json schema and json validation. It could look like this:

```
schema should validateJson (json)
```

and when the validation fails, we should see a message which tells us sth like this:
"JSON [XYZ] cannot be validated by schema [ZYX]"

Looking at this error you will be able to say why (at least you have all the data to analyze this case). 
In the test below, you will get sth like this: "should be Success(()), but there was Failure(ex)`. To figure out what went wrong, you will be forced to run the code and use a debugger (or dupa-debugging method).